tests/test_packages/test_skills/test_generic_buyer/test_behaviours.py

```diff
@@ -0,0 +1,329 @@
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains test case classes based on pytest for AEA end-to-end testing."""
+
+from pathlib import Path
+from typing import cast
+
+from aea.helpers.search.models import Description
+from aea.protocols.default.message import DefaultMessage
+from aea.protocols.dialogue.base import DialogueMessage
+from aea.test_tools.test_skill import BaseSkillTestCase
+
+from packages.fetchai.connections.ledger.base import CONNECTION_ID as LEDGER_PUBLIC_ID
+from packages.fetchai.protocols.fipa.message import FipaMessage
+from packages.fetchai.protocols.ledger_api.message import LedgerApiMessage
+from packages.fetchai.protocols.oef_search.message import OefSearchMessage
+from packages.fetchai.skills.generic_buyer.behaviours import GenericSearchBehaviour
+from packages.fetchai.skills.generic_buyer.dialogues import FipaDialogue, FipaDialogues
+from packages.fetchai.skills.generic_buyer.handlers import GenericFipaHandler
+from packages.fetchai.skills.generic_buyer.strategy import GenericStrategy
+
+from tests.conftest import ROOT_DIR
+
+
+FETCHAI = "fetchai"
+
+
+class TestSkillBehaviour(BaseSkillTestCase):
+    """Test behaviours of generic buyer."""
+
+    path_to_skill = Path(ROOT_DIR, "packages", "fetchai", "skills", "generic_buyer")
+
+    @classmethod
+    def setup(cls):
+        """Setup the test class."""
+        super().setup()
+        cls.search_behaviour = cast(
+            GenericSearchBehaviour, cls._skill.skill_context.behaviours.search
+        )
+
+    def test_search_behaviour_setup_is_ledger_tx(self):
+        """Test the setup method of the search behaviour where is_ledger_tx is True."""
+        # operation
+        self.search_behaviour.setup()
+
+        # after
+        assert self.get_quantity_in_outbox() == 1, "No message in outbox."
+        has_attributes, error_str = self.message_has_attributes(
+            actual_message=self.get_message_from_outbox(),
+            message_type=LedgerApiMessage,
+            performative=LedgerApiMessage.Performative.GET_BALANCE,
+            to=str(LEDGER_PUBLIC_ID),
+            sender=self.skill.skill_context.agent_address,
+            ledger_id=FETCHAI,
+            address=self.skill.skill_context.agent_address,
+        )
+        assert has_attributes, error_str
+
+    def test_search_behaviour_setup_not_is_ledger_tx(self):
+        """Test the setup method of the search behaviour where is_ledger_tx is False."""
+        # setup
+        strategy = cast(GenericStrategy, self.skill.skill_context.strategy)
```

Comment:
could also be set in setup, no real cost if we don't use it..and saves some loc

Reply:
Yes